### ISSUE TITLE
Build products should not go in the Drake install tree except during packaging

### ISSUE THREAD

**Issue by david-german-tri** [RobotLocomotion/drake#4917]
We don't know what, if anything, still depends on this, so some research is required.

**Comment by david-german-tri** [comment]
This will be a moot point after #3129, since Bazel allows only declared dependencies.  Moreover, the existence of a Bazel build covering almost our entire tree proves that our actual C++ deps are already sane. Closing as wontfix.
